documentation/List-of-Action-plugins.md

```diff
@@ -406,6 +408,73 @@ The list of changed review/responses submitted is passed as `changedResponses` t
+| ---------------------------------------- | ------------------------------------------ |
+| `docTemplateId`\*                        | `document: {uniqueId, filename, filepath}` |
+| `data`                                   |                                            |
+| `additionalData`                         |                                            |
+| `userId`                                 |                                            |
+| `applicationSerial`                      |                                            |
+
+The Action utilises the internal `generatePDF` function, which is also accessible via the [`/generate-pdf` endpoint](API.md)
+
+`docTemplateId` specifies the uniqueId of the carbone template file (from the "file" table)
+
+`data` is the object that contains all the data for substituting values into the carbone template. You can specifify a `data` object explicitly, or by default it will use properties `applicationData` and `outputCumulative`. If you want to _add_ to these, you can provide an `additionalData` parameter object which will be added as an additional field, like so:
+
+```
+data: { ...applicationData, ...outputCumulative, additionalData }
+```
+
+`userId` and `applicationSerial` are not required for functioning, but will be stored as fields in the resulting "file" record. If `applicationSerial` is supplied, the output PDF file will be stored in a subfolder named for the application serial.
+
+The output object `document` provides the `uniqueId`, `filename`, and `filepath` (relative to server root) of the generated document, so it can be accessed by subsequent actions
+
+---
+
+### Send Notification
+
+Generates notifications and sends email. For now, there is no UI for notifications in the app, so emails are the primary means of notifying users. A notification record is stored in the database though ("notification" table).
+
+- _Action Code:_ **`sendNotification`**
+
+| Input parameters<br />(\*required) <br/> | Output properties |
+| ---------------------------------------- | ----------------- |
+| `subject`\*                              | `notification`    |
+| `message`\*                              |                   |
+| `email`                                  |                   |
+| `userId`                                 |                   |
+| `fromName`                               |                   |
+| `fromEmail`                              |                   |
+| `attachments`                            |                   |
+| `sendEmail`                              |                   |
+
+This action requires the use of an external SMTP server for sending the emails. You'll need to provide configuration details for this in the `config.json` file within this actions folder. The SMTP password needs to be stored in the systems (non-shared) `.env` or passed in as an argument at runtime (when running on production server)
```

Comment:
It might be good to show the exact format expected in `.env`:
```
SMTP_PASSWORD=<secret>
```

Reply:
Yeah, good idea: 9cad6035